libs/sdk-ui-ext/src/internal/dashboardEmbedding/mergeFiltersWithDashboard.ts

```diff
@@ -0,0 +1,19 @@
+// (C) 2020-2021 GoodData Corporation
+import { IDashboard, FilterContextItem } from "@gooddata/sdk-backend-spi";
+import { IDashboardFilter } from "./types";
+import { filterArrayToFilterContextItems } from "./utils/filters";
+
+/**
+ * Gets filters merged with the filters provided by the specified dashboard.
+ *
+ * @param dashboard - dashboard to get the filters from
+ * @param additionalFilters - filters to apply on top of the filters from the dashboard
+ * @alpha
+ */
+export function mergeFiltersWithDashboard(
+    dashboard: IDashboard,
```

Comment:
I'm just thinking about signature of this function, and I'm not sure that we want to bother user with FilterContextItem(s), is there some use case for this? + He is forced to use useDashboard hook with this...
Just one idea, what about hook accepting only IDashboardFilter + dashboard ObjRef ?
Maybe some live-example code representing this merging use case could be useful for us to make a "bigger picture" about it.

Reply:
There is a usecase for this, unfortunately: if they want to replicate KPI Dashboards behavoiur that a date filter filters *every* date dimension the same way. The normal `IDateFilter` must have a date dimension specified, `IDashboardDateFilter` does not and the semantics are so that it filters all the dimensions.

I chose this signature for convenience, the user gets the `IDashboard` object with the `onDashboardLoaded` and they can simply pass it to this function (without having to worry where to get the filterContext items). So they do not have to use the hook, the callback is enough here.